Source/Core/Common/JsonUtil.h

```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <span>
+
+#include <picojson.h>
+
+template <typename T, std::size_t N>
+picojson::array ToJsonArray(const std::array<T, N>& value)
```

Comment:
what does the compiler error out with when this is `std::span<const T>` and the `N` template parameter is removed?

Reply:
```
'picojson::array ToJsonArray(const std::span<T,18446744073709551615> &)': could not deduce template argument for 'const std::span<T,18446744073709551615> &' from 'const std::array<float,3>'
ShaderAsset.cpp(352,65): message :                      json_property["default"] = picojson::value{ToJsonArray(default_value.value)};
```

Similar error without the `const&` and making `std::span<const T>`.  Can't deduce template argument.

Reply:
Oh, right, implicit conversions aren't considered in function template deduction. The only real way around it is to either leave it as an array, or just generically accept anything that can be iterated on, which is the most flexible:

```cpp
// Ideally this would use a concept like, 'template <std::ranges::range Range>' to constrain it,
// but unfortunately we'd need to require clang 15 for that, since the ranges library isn't
// fully implemented until then, but this should suffice.

template <typename Range>
picojson::array ToJsonArray(const Range& data)
{
  picojson::array result;
  result.reserve(std::size(data));

  for (const auto& value : data)
  {
    result.emplace_back(static_cast<double>(value));
  }

  return result;
}
```

Reply:
Ok, I did consider that but wasn't sure if I liked it.  I guess `data` will throw compiler errors if it isn't iteratable or able to get size.  I'll make that change.